create_openapi_definition.py

```diff
@@ -0,0 +1,213 @@
+#               "href": "/tickets"
+#           },
+#           "comments": {
+#               "href": "/comments"
+#           },
+#           "attachments": {
+#               "href": "/attachments"
+#           },
+#           "customer_user_associations": {
+#               "href": "/customer_user_associations"
+#           }
+#       }
+#   }
+yaml = YAML()
+
+if __name__ == "__main__":
+    # Recursively go through the openapi_resource_definitions directory to get each resource's
+    # OpenAPI definition.
+    for dirpath, dirnames, files in os.walk("openapi_resource_definitions"):
+        # If dirnames is an empty list, it means we are at the end of a branch in the directory
+        # tree, which means the yaml files for one of the resources should be in this
+        # directory.
+        if not dirnames:
+            # Get the name of the resource that we're processing.
+            # dirpath will look something like:
+            #
+            # openapi_resource_definitions/individual_resources/Attachment
+            #
+            # So the resource name will be the last item of dirpath.
+            resource_name = os.path.split(dirpath)[-1]
+            # Iterate through the list of definition files for each HTTP method inside that
+            # resource's folder.
+            for def_file in files:
+                # Get the path to the definition file.
+                path_to_file = os.path.join(dirpath, def_file)
+                # Get the type of HTTP method from the file name by removing the ".yaml" file
+                # extension from the string.
+                http_method = def_file[:-5]
+                # File names for our resource definitions should be in the form of:
+                # "get.yaml"
+                # "post.yaml"
+                # or some other RESTful HTTP method that we support in our API. If the current
+                # file being processed isn't like that, then we skip over that file since it's
+                # not a resource definition.
+                if http_method not in ALLOWED_HTTP_METHODS:
+                    continue
+                # Load the yaml data from the current file being processed
+                with open(path_to_file, "r") as yaml_file:
+                    definition        = yaml.load(yaml_file)
+                    resource_url_path = RESOURCE_PATH_LOOKUP[resource_name]
+                    # Set the value of the resource path in the base dictionary to the
+                    # definition
+                    # we loaded from the yaml file. It will look something like this in the
+                    # paths dictionary:
+                    #
+                    # paths:
+                    #   /:
+                    #     get:
+                    #       ...
+                    openapi_definition['paths'][resource_url_path][http_method] = definition
+
+    # Get the date and time for right now, then generate a name for the definition file
```

Comment:
A better comment here might be: Why are we using `now` as part of the filename? I can see that it's used, but I don't know the reason behind it.

Reply:
I just used `datetime.now()` kindof as a placeholder, as in "I created this definition file at this point in time" (much like a log file or report). I did it this way because there are times where we may need to tweak definitions, so keeping files separated by timestamp means we can revert back to a previous version if there are any mistakes in the newer generated file (which may inadvertently be overwritten both here and on Swaggerhub)

Reply:
Yes, yes, I get that. I meant that it should be documented in the code comment.

Reply:
Ah right, I see. Okay, I added a comment explaining why we use the date and time